File: spring-java-starter/spring-java-mysql/starter/src/main/java/io/github/mrgsrylm/repository/RefreshTokenRepository.java
package io.github.mrgsrylm.repository;

import io.github.mrgsrylm.entity.RefreshTokenEntity;
import io.github.mrgsrylm.entity.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Modifying;

import java.util.Optional;

public interface RefreshTokenRepository extends JpaRepository<RefreshTokenEntity, Long> {
    RefreshTokenEntity findByUserId(Long userId);
    Optional<RefreshTokenEntity> findByToken(String token);
    @Modifying
    int deleteByUser(UserEntity user);
}

File: spring-kotlin-starter/spring-kotlin-mysql/starter/src/main/kotlin/io/github/mrgsrylm/repository/RefreshTokenRepository.kt
package io.github.mrgsrylm.repository

import io.github.mrgsrylm.entity.RefreshTokenEntity
import io.github.mrgsrylm.entity.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Modifying

interface RefreshTokenRepository : JpaRepository<RefreshTokenEntity, Long> {
    fun findByUserId(userId: Long): RefreshTokenEntity
    fun findByToken(token: String): RefreshTokenEntity?

    @Modifying
    fun deleteByUser(user: UserEntity): Int
}